src/main/groovy/com/okta/test/mock/application/ApplicationTestRunner.groovy

```diff
@@ -70,23 +73,20 @@ abstract class ApplicationTestRunner extends HttpMock {
         return app.getTestScenario().loginRedirectPath
     }
 
-    @BeforeMethod
-    void checkToRun(Method method) {
-        if (scenario.enabled == false) {
-            throw new SkipException("Skipping the disabled scenario")
-        }
+    @Override
+    void run(IHookCallBack callBack, ITestResult testResult) {
 
-        for (String disabledTest : scenario.disabledTests) {
-            if (method.getName().equals(disabledTest)) {
-                throw new SkipException("Skipping the disabled test - " + disabledTest)
-            }
+        String testMethod = testResult.method.getMethodName()
+        if (scenario.enabled == false || scenario.disabledTests.contains(testMethod)) {
```

Comment:
nit: Can you use negation instead?

```java
if (!scenario.enabled || scenario.disabledTests.contains(testMethod)) {
  ...
}
```

Reply:
scenario.enabled is a `Boolean` not a `bool` so that wouldn't work as it could create an NPE if `enabled` was `null`